pkg/stub/handler.go

```diff
@@ -304,6 +305,48 @@ func (h *Handler) updateCfgArch(cfg *v1.Config) *v1.Config {
 	return cfg
 }
 
+//TODO leaving this in for now to promote discussion of this which started in planning and enhancement proprosal;
+// can delete if need be as PR review process completes
+func (h *Handler) Disconnected() bool {
+	if h.configclient == nil {
+		// unit test environment
+		return false
+	}
+	err := wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
+		// if not set, unit test environment, so don't bother
+		proxy, err := h.configclient.Proxies().Get("cluster", metav1.GetOptions{})
+		if err == nil &&
+			(len(proxy.Status.HTTPProxy) > 0 || len(proxy.Status.HTTPSProxy) > 0) {
+			// if proxies are involved we assume we are not in a disconnected scenario
+			return true, nil
+		}
+
+		tlsConf := &tls.Config{}
+		conn, err := tls.Dial("tcp", "registry.redhat.io:443", tlsConf)
+		if err != nil {
+			logrus.Infof("test connection to registry.redhat.io failed with %s", err.Error())
+			return false, nil
+		}
+		defer conn.Close()
+		err = conn.Handshake()
+		if err != nil {
+			logrus.Infof("test connection to registry.redhat.io experienced SSL handshake error %s", err.Error())
+			// these can be intermittent as well so we'll retry
+			return false, nil
+		}
+		logrus.Infof("test connection to registry.redhat.io successful")
+		return true, nil
+
+	})
+
+	if err == nil {
+		return false
+	}
+
+	logrus.Infof("unable to establish HTTPS connection to registry.redhat.io after 1 minute, bootstrap to Removed")
+	return false
```

Comment:
or this should be true (since the method name is "Disconnected", returning false implies we are *not* disconnected)

Reply:
ah forgot about the name of the method (was originally just a placeholder)

will address momentarily

Reply:
method name addressed @bparees PTAL

Reply:
you still need to change this to return `true` right?

Reply:
duh ... yes ... just a moment

Reply:
ok no returning true thanks @bparees